internal/runners/activate/activate.go

```diff
@@ -116,6 +124,32 @@ func (r *Activate) run(params *ActivateParams, activatorLoop activationLoopFunc)
 	return activatorLoop(r.out, r.subshell, projectPath, activate)
 }
 
+func updateProject(prjFile *projectfile.Project, names *project.Namespaced) (*project.Project, error) {
```

Comment:
This shouldn't be returning the project. That's making this function do more than it's tasked to.